cmd/ndm_daemonset/probe/uuid.go

```diff
@@ -84,6 +86,28 @@ func generateUUID(bd blockdevice.BlockDevice) (string, bool) {
 		klog.Infof("device(%s) has a filesystem, using filesystem UUID: %s", bd.DevPath, bd.FSInfo.FileSystemUUID)
 		uuidField = bd.FSInfo.FileSystemUUID
 		ok = true
+	case features.FeatureGates.IsEnabled(features.PartitionTableUUID) && len(bd.PartitionInfo.PartitionTableType) > 0:
+		// if the host is CentOS 7, the `libblkid` version on host is `2.23`,
+		// but the `PTUUID` tag was start to provide from `2.24`. This will cause
+		// the udev cache fetched from host udevd will not contain env `ID_PART_TABLE_UUID`.
+		// Fortunately, the `libblkid` version in our base container (ubuntu 16.04)
+		// is `2.27.1`, will provide `PTUUID` tag, so we should fetch `PTUUID` from `blkid`.
+		partUUID := ""
+		if len(bd.PartitionInfo.PartitionTableUUID) == 0 {
+			deviceIdentifier := blkid.DeviceIdentifier{DevPath: bd.DevPath}
```

Comment:
Instead of fetching this data here, can we fill this data from some probes. The approach that we follow is all the data should be made available via the various probes. The handler should be just processing the data.